modules/dbparser/groupingby.c

```diff
@@ -272,8 +269,16 @@ grouping_by_expire_entry(TimerWheel *wheel, guint64 now, gpointer user_data)
   if (self->sort_key_template)
     correllation_context_sort(context, self->sort_key_template);
   LogMessage *msg = grouping_by_emit_synthetic(self, context);
-  if (msg)
-    log_msg_unref(msg);
+  if (!msg)
```

Comment:
minor: I prefer having positive branches prior to warnings. Error handling would be different though.

Reply:
agree, additionally I've dropped the patch that extracted the debug msg (it is better imho down there).